sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2644,42 +2688,63 @@ class AstBuilder extends SqlBaseParserBaseVisitor[AnyRef] with SQLConfHelper wit
   }
 
   /**
-   * Create an [[UnresolvedRelation]] from a multi-part identifier context.
+   * Create an [[UnresolvedRelation]] from a multi-part identifier.
    */
   private def createUnresolvedRelation(
-      ctx: MultipartIdentifierContext): UnresolvedRelation = withOrigin(ctx) {
-    UnresolvedRelation(visitMultipartIdentifier(ctx))
+      ctx: ParserRuleContext, ident: Seq[String]): UnresolvedRelation = withOrigin(ctx) {
+    UnresolvedRelation(ident)
```

Comment:
shall we take `IdentifierReferenceContext` as input and make this function return `LogicalPlan`? Then we can create `PlanWithIdentifier` here.

Reply:
Yes, I did wonder about that

Reply:
Actually MOST of the uses of createRelation() cannot take withIdentClause immediately around UnresolvedRelation.
See my comments on INSERT INTO